ttnn/cpp/ttnn/operations/data_movement/reshape_on_device/reshape.hpp

```diff
@@ -14,21 +14,19 @@ struct ReshapeOperation {
     static ttnn::Tensor invoke(
         uint8_t queue_id,
         const ttnn::Tensor& input_tensor,
-        int N,
-        int C,
-        int H,
-        int W,
+        ttnn::Shape shape,
         const std::optional<MemoryConfig>& memory_config_arg);
 
     static ttnn::Tensor invoke(
         const ttnn::Tensor& input_tensor,
-        int N,
-        int C,
-        int H,
-        int W,
+        ttnn::Shape shape,
         const std::optional<MemoryConfig>& memory_config);
 
-    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, int N, int C, int H, int W);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::Shape& shape);
+
+    static ttnn::Tensor invoke(uint8_t queue_id, const ttnn::Tensor& input_tensor, const std::vector<int32_t>& shape_vector, const std::optional<MemoryConfig>& memory_config_arg);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const std::vector<int32_t>& shape_vector, const std::optional<MemoryConfig>& memory_config_arg);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const std::vector<int32_t>& shape_vector);
```

Comment:
Why do we need new overloads? Also, can we use `ttnn::SimpleShape` instead of `std::vector<int32_t>` to represent the input type?

Reply:
`std::vector<int32_t>` has a different meaning from SimpleShape. `std::vector<int32_t>` allows for a single `-1` within the values, meaning - "calculate this dimension to match the volume"

Reply:
ah ok